src/array_dist.jl

```diff
@@ -0,0 +1,90 @@
+# Univariate
+
+const VectorOfUnivariate{
+    S <: ValueSupport,
+    Tdist <: UnivariateDistribution{S},
+    Tdists <: AbstractVector{Tdist},
+} = Distributions.Product{S, Tdist, Tdists}
+
+function ArrayDist(dists::AbstractVector{<:Normal{T}}) where {T}
+    if T <: TrackedReal
+        init_m = vcat(dists[1].μ)
+        means = mapreduce(vcat, drop(dists, 1); init = init_m) do d
+            d.μ
+        end
+        init_v = vcat(dists[1].σ^2)
+        vars = mapreduce(vcat, drop(dists, 1); init = init_v) do d
+            d.σ^2
+        end
+    else
+        means = [d.μ for d in dists]
+        vars = [d.σ^2 for d in dists]
+    end
+
+    return MvNormal(means, vars)
+end
+function ArrayDist(dists::AbstractVector{<:UnivariateDistribution})
```

Comment:
`product_distribution` is too long. I don't really like the name. `arraydist` is fine I think.

Reply:
Sure, I'm fine with `arraydist` and I don't think we want users to use `product_distribution` since then they can't take advantage of all the dispatches that we define here. My only suggestion is to define
```julia
arraydist(dists::AbstractVector{<:UnivariateDistribution}) = product_distribution(dists)
```
since then you neither have to deal with the standard case for `MvNormal` above (you only define `arraydist(dists::AbstractVector{<:Normal{<:TrackedReal}}))` and even more importantly if any other nice simplifications would be defined for `product_distribution` (in Distributions or some other package) `arraydist` would take advantage of them automatically (that's not the case if you just define `arraydist(dists::AbstractVector{<:UnivariateDistribution}) = Product(dists)`).

Sorry if that was not clear from my comment.

Reply:
Done